src/pages/fi-dim-values/index.js

```diff
@@ -0,0 +1,179 @@
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+  const [tpaValues, setTpaValues] = useState([])
+  const [error, setError] = useState(false)
+
+  const [selectedTpaValue, setSelectedTpaValue] = useState({ key: null, value: '' })
+  const formatedRecordId = typeof selectedTpaValue?.key == 'string' ? selectedTpaValue.key.match(/\d+/)?.[0] : null
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    refetch,
+    invalidate,
+    filterBy,
+    clearFilter,
+    access
+  } = useResourceQuery({
+    datasetId: ResourceIds.DimensionsValues,
+    filter: {
+      endpointId: FinancialRepository.DimensionValue.qry,
+      filterFn: fetchWithSearch
+    }
+  })
+
+  useEffect(() => {
+    if (formatedRecordId !== null) {
+      filterBy('qry', formatedRecordId)
+    } else {
+      clearFilter('qry')
+    }
+  }, [formatedRecordId])
+
+  async function fetchWithSearch({ filters }) {
+    if (formatedRecordId) {
+      const data = await getRequest({
+        extension: FinancialRepository.DimensionValue.qry,
+        parameters: `_filter=${filters.qry}&_dimension=${formatedRecordId}`
+      })
+
+      return data
+    }
+  }
+
+  const columns = [
+    {
+      field: 'id',
+      headerName: _labels.id,
+      flex: 1
+    },
+    {
+      field: 'name',
+      headerName: _labels.name,
+      flex: 1
+    }
+  ]
+
+  const del = async obj => {
+    try {
+      await postRequest({
+        extension: FinancialRepository.DimensionValue.del,
+        record: JSON.stringify(obj)
+      })
+      invalidate()
+      toast.success(platformLabels.Deleted)
+    } catch (error) {
+      toast.error(platformLabels.Error)
+    }
+  }
+
+  const add = () => {
+    if (formatedRecordId) {
+      openForm()
+    } else {
+      setError(true)
+    }
+  }
+
+  const edit = obj => {
+    openForm(obj?.id)
+  }
+
+  function openForm(id) {
+    stack({
+      Component: DimValuesForm,
+      props: {
+        labels: _labels,
+        id: id,
+        maxAccess: access,
+        invalidate,
+        dimensionId: formatedRecordId
+      },
+      width: 600,
+      height: 500,
+      title: _labels.dimValues
+    })
+  }
+  const emptyValues = item => item.value !== null && item.value !== ''
+
+  useEffect(() => {
+    ;(async () => {
+      const data = await getRequest({
+        extension: SystemRepository.Defaults.qry,
+        parameters: `_filter=tpaDimension`
+      })
+
+      if (data) {
+        const result = data.list.filter(emptyValues)
+        setTpaValues(result)
+        setSelectedTpaValue(result[0])
+        setError(false)
+      }
+    })()
+  }, [])
+
+  return (
+    <VertLayout>
+      <Fixed>
+        <Grid container alignItems='center' spacing={2}>
+          <Grid item>
+            <GridToolbar onAdd={add} maxAccess={access} labels={_labels} />
+          </Grid>
+          <Grid item xs={4}>
+            <ResourceComboBox
```

Comment:
If you're not passing resourceId to resourcecombobox use customcombobox. Make sure that tpavalues can't be fetched from the api independently before doing that.